presto-main/src/main/java/com/facebook/presto/sql/planner/SubqueryPlanner.java

```diff
@@ -264,18 +264,21 @@ private PlanBuilder appendExistSubqueryApplyNode(PlanBuilder subPlan, ExistsPred
 
         PlanBuilder subqueryPlan = createPlanBuilder(existsPredicate.getSubquery());
 
-        if (isAggregationWithEmptyGroupBy(subqueryPlan.getRoot())) {
+        PlanNode subqueryPlanRoot = subqueryPlan.getRoot();
+        if (isAggregationWithEmptyGroupBy(subqueryPlanRoot)) {
             subPlan.getTranslations().put(existsPredicate, BooleanLiteral.TRUE_LITERAL);
             return subPlan;
         }
 
         Symbol exists = symbolAllocator.newSymbol("exists", BOOLEAN);
         subPlan.getTranslations().put(existsPredicate, exists);
+        ExistsPredicate rewrittenExistsPredicate = new ExistsPredicate(
```

Comment:
Some issues that occurred to me in this PR @martint (I approved it though).

Can it happen that there is no subquery output symbol during planning?

I am not sure exists should care about symbols. It seems that exists should care about cardinality only. You can have non-zero cardinality with empty rows.

This will limit `PruneUnreferencedOutputs` functionality on the randomly selected symbol: `subqueryPlanRoot.getOutputSymbols().get(0)`. The symbols should not be removed since it's referenced by `ExistsPredicate`. However, it's not needed to rewrite exists into count aggregation.